remappers/hostmetrics/metric.go

```diff
@@ -0,0 +1,75 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
```

Comment:
Any reason why we can't use the existing metric.go file ?
The code looks very similar with minor changes. 
Also we would like to have the attributes in the structure as in the network, disk, process scraper etc. we would be using them. So, I think its prudent to edit in the existing metric.go file

Reply:
> Any reason why we can't use the existing metric.go file ?

It is how git shows the diff when I moved the file, I have not actually removed it.

> Also we would like to have the attributes in the structure as in the network, disk, process scraper etc. we would be using them.

How exactly would the `addMetric` use them, is the `mutator` logic not good enough? The [current code](https://github.com/elastic/opentelemetry-lib/blob/fdb46e794d6b6b95b2edf5f1fbe43e423e4eed2e/hostmetrics/metric.go#L53-L57) in the metric uses conditionals to add process-specific attributes which leaks dataset specific logic in the more generic `addMetric` function. The PR introduces a `mutator` func which you can pass in to add any scraper-specific attributes/logic.